src/Table/Table.tsx

```diff
@@ -0,0 +1,985 @@
+
+export type DataItem = { [name: string]: any } & TableRow;
+type RowTypes = "row" | "subRow";
+
+export interface Column {
+    label: string;
+    accessor: string;
+    canSort?: boolean;
+}
+
+export interface ActionLinkItem {
+    label: string;
+    onClick: (event: React.MouseEvent<HTMLAnchorElement, MouseEvent>, selectedRow: TableRow) => void;
+}
+
+export interface PrimaryActionButton {
+    label: string;
+    onClick: (event: React.MouseEvent<HTMLButtonElement, MouseEvent>, selectedRow: TableRow) => void;
+}
+
+export interface TableHeader extends Column {
+    isSorted?: boolean;
+    isSortedDesc?: boolean;
+}
+
+interface Cell {
+    id: string | number;
+    accessor: string;
+    value: string | number | boolean;
+}
+
+export interface TableRow {
+    rowIndex: number;
+    cells: Array<Cell>;
+    selected?: boolean;
+    actionsDropdownDropped?: boolean;
+    subRows?: Array<TableRow>;
+    expanded?: boolean;
+    rowContentDetail?: React.ReactNode;
+}
+
+export const enum sortDirectionTypes {
+    Ascending = "ASC",
+    Descending = "DESC"
+}
+
+/**
+ * sum the total of columns or cols in a row
+ * @param colsLength the length of the columns
+ * @param useSelection add a column for selection checkboxes
+ * @param useShowActionColumn add another column for action columns
+ * @param useGroupBy add another columns for groupby
+ */
+function sumCols(colsLength: number, useSelection?: boolean, useShowActionColumn?: boolean, useGroupBy?: boolean): number {
+    let sum = colsLength;
+    if (useSelection || useGroupBy) {
+        if (useSelection) {
+            sum = sum + 1;
+        }
+
+        if (useGroupBy) {
+            sum = sum + 1;
+        }
+        if (useShowActionColumn) {
+            sum = sum + 1;
+        }
+    }
+
+    return sum;
+}
+
+/**
+ * sort array of tabke rows
+ * @param items table rows array
+ * @param columnName the target column name
+ * @param sortDirection the sort direction
+ */
+function sortArray(items: Array<TableRow> = [], columnName: string, sortDirection: sortDirectionTypes): Array<TableRow> {
+    const sortedItems: Array<any> = [...items].sort((firstItem: TableRow, secondItem: TableRow) => {
+        let result: number = 0;
+        if (sortDirection === sortDirectionTypes.Ascending) {
+            if (isNaN(secondItem[columnName]) && isNaN(firstItem[columnName])) {
+                result = String(firstItem[columnName]).localeCompare(String(secondItem[columnName]), ["sw", "en"], { sensitivity: "base", ignorePunctuation: true });
```

Comment:
Why is Swedish hardcoded here? this component is open source. What if the user is from German. Maybe should utilize `navigator.languages` or allow the user to pass the locale

Reply:
I don't recommend exposing the locale as an input or params for users to pass. I feel like it will raise a lot of questions on what does it do and what is it for. It appearance in this domain will send a wrong signal to the user, thinking you can handle some kind of translation or language service for him. However, i do agree on using language locale and having a default where not available.